workflow/automation/lib/schedulers/abstractscheduler.py

```diff
@@ -70,6 +70,15 @@ def check_queues(self, user: bool = False, target_machine=None) -> List[str]:
         """
         pass
 
+    @abstractmethod
+    def check_wct(self, job_id: int) -> bool:
```

Comment:
Name should be 'check_wct_hit' or something and the return should line up with the function name
I also think there should be a third state for undetermined, but None is already a bit loaded as a return value